x-pack/plugins/data_visualizer/public/application/index_data_visualizer/hooks/use_field_stats.ts

```diff
@@ -184,41 +200,47 @@ export function useFieldStatsSearchStrategy(
       });
     };
 
+    const statsMapTmp = new Map<string, FieldStats>();
+
     // First, attempt to fetch field stats in batches of 10
-    searchSubscription$.current = fieldStatsSub.subscribe({
-      next: (resp) => {
-        if (resp) {
-          const statsMap = new Map<string, FieldStats>();
-          const failedFields: Field[] = [];
-          resp.forEach((batchResponse) => {
+    searchSubscription$.current = from(fieldStatsToFetch)
```

Comment:
Using the 1.6M doc `siem-auditbeat` dataset, which has a lot of non-populated fields, if I enable 'Show empty fields', the distribution data for the rows and expanded content will never load up, and just shows the loading spinner. This isn't happening on `main`.

![image](https://user-images.githubusercontent.com/7405507/159283070-26543219-2dcb-42a6-bdf7-3d04b0f36d0c.png)